File: test2.lua
io.write("hello!\n")
io.write("Please Input a number: ")
local Input = tonumber(io.read())
if not Input then return end
io.write("\n")
local A = Input%2
io.write("The number is ")
if A == 0 then
  io.write("Even\n")
elseif A == 1 then
  io.write("Odd\n")
else
  io.write("Decimal,Remainder: " ..  A .. "\n")
end
io.write("The number has ")
if Input > 0 then
  local B = math.sqrt(Input)
  io.write("two roots: " .. B .. "and" .. -B  .."\n")
elseif Input == 0 then
  local B = 0
  io.write("one root: " .. B .. "\n")
else 
  local B = math.sqrt(-Input)
  io.write("no real root\n")
  io.write("Two of Imaginary roots: " .. B .. "i and " .. -B .. "i\n")
end

File: test2.py
import math
print("Hello!\n")
try:
    Input = float(input("Please enter a number: "))
except ValueError:
    print("Invalid input!")
    exit()

A = round(Input%2,10)
if A == 0:
  print(f"The number is Even\n")
elif A == 1:
  print(f"The number is Odd\n")
else:
  print(f"The number is Decimal,Remainder: {A}\n")

if Input > 0:
  rt = math.sqrt(Input)
  print(f"The number has two roots: {rt} and {-rt}")
elif Input == 0:
  print(f"The number has one root: 0")
else:
  print(f"The number doesnt have real root\n")
  rt = math.sqrt(-Input)
  print(f"Two of Imaginary roots: {rt}i and {-rt}i")
